src/symbol/projection.js

```diff
@@ -183,39 +187,41 @@ function getGlCoordMatrix(posMatrix: Float32Array,
     }
 }
 
-function project(point: {x: number, y: number}, matrix: Mat4, elevation: number = 0): ProjectedSymbol {
-    const pos = [point.x, point.y, elevation, 1];
+function project(x: number, y: number, matrix: Mat4, elevation: number = 0): Vec4 {
+    const pos = [x, y, elevation, 1];
     if (elevation) {
         vec4.transformMat4(pos, pos, matrix);
     } else {
         xyTransformMat4(pos, pos, matrix);
     }
     const w = pos[3];
-    return {
-        point: [pos[0] / w, pos[1] / w, pos[2] / w],
-        signedDistanceFromCamera: w
-    };
+    pos[0] /= w;
+    pos[1] /= w;
+    pos[2] /= w;
+    return pos;
 }
 
-function projectVector(point: [number, number, number], matrix: Mat4): ProjectedSymbol {
+function projectVector(point: Vec3, matrix: Mat4): Vec4 {
     const pos = [point[0], point[1], point[2], 1];
     vec4.transformMat4(pos, pos, matrix);
     const w = pos[3];
-    return {
-        point: [pos[0] / w, pos[1] / w, pos[2] / w],
-        signedDistanceFromCamera: w
-    };
+    pos[0] /= w;
+    pos[1] /= w;
+    pos[2] /= w;
+    return pos;
```

Comment:
Interesting! We could inline in some cases, but I see there's at least one use where it needs the `w` component afterwards, so this could be non-trivial.

Reply:
Yes, looking at the PR more, it seems like we need the `w` value in the [callee side](https://github.com/mapbox/mapbox-gl-js/pull/12351/files#diff-11d2278294f32285af99803c1ef79ee58bd1845a23050a627f1dada8ba8b4d54R333) in a few places (for both `project` and `projectVector`), so probably not feasible and ok as is.

Reply:
I ended up removing `projectVector` and updating the code to use `project`.